src/utils/__tests__/safeRename.spec.js

```diff
@@ -0,0 +1,79 @@
+import childProcess from 'child_process';
+import fsPromises from 'fs/promises';
+import fs from 'fs';
+import os from 'os';
+import { join, resolve } from 'path';
+import { remove as extraRemove } from 'fs-extra';
+
+import safeRename from '../safeRename.js';
+
+const assetsDir = resolve('./src/utils/__tests__/assets/safeRename');
+
+/* input tar file contents:
+ * rename1.txt
+ * a/
+ *   a1.txt
+ *   a2.txt
+ *   b/
+ *     b1.txt
+ */
+
+const mockFunc = jest.fn().mockImplementation(() => {
+  throw new Error('EXDEV: cross-device link not permittted');
+});
+
+fsPromises.rename = mockFunc;
+
+function fileExists(path) {
+  try {
+    fs.accessSync(path, fs.constants.F_OK);
+
+    return true;
+  } catch (_) {
+    return false;
+  }
+}
+
+describe('safeRename', () => {
+  let targetDir;
+
+  beforeEach(() => {
+    childProcess.execFileSync('tar', ['xf', join(assetsDir, 'safeRename.tar')], { cwd: assetsDir });
+    targetDir = fs.mkdtempSync(join(os.tmpdir(), 'rename-'));
+  });
+  afterEach(async() => {
+    // cleanup
+    for (const entry of [targetDir, 'rename1.txt', 'a']) {
+      try {
+        const fullPath = entry[0] === '/' ? entry : join(assetsDir, entry);
+
+        if (fileExists(fullPath)) {
+          await extraRemove(fullPath);
```

Comment:
Is `fsPromises.rm(fullPath, {recursive: true})` insufficient?

Reply:
As I wrote somewhere else here, `rmdir` works for directories.